### PR TITLE
Write new IMU integrator

### PR DESCRIPTION
**Pull request by JeffreyZh4ng**
Because licenses.

### CODE REVIEW

File: imu_integrator/plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+		// if (input_values->last_cam_integration_time > last_cam_time) {
+		// 	cam_count++;
+		// 	last_cam_time = input_values->last_cam_integration_time;
+		// 	std::cout << "Num IMUs recieved since last cam: " << counter << " Diff between new cam and latest IMU: " 
+		// 			  << timestamp - last_cam_time << " Expected IMUs recieved VS Actual: " << cam_count*10 << ", " << total_imu << std::endl;
+		// 	counter = 0;
+		// }
+		// counter++;
+		if (input_values->last_cam_integration_time > last_cam_time) {
+			std::cout << "&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&\n";
```
Comment (charmoniumQ) [inline imu_integrator/plugin.cpp:140]: Remove the couts before merging. You can guard them with `#ifndef NDEBUG` or remove them.

File: imu_integrator/plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+			bias.gyroscope(),
+			bias.accelerometer(),
+			out_pose.translation(), // Position
+			navstate_k.velocity(), // Velocity
+			out_pose.rotation().toQuaternion(), // Eigen Quat
+			real_time
+		});
+	}
+
+	std::vector<imu_type> select_imu_readings(const std::vector<imu_type>& imu_data, double time_begin, double time_end) {
```
Comment (mhuzai) [inline imu_integrator/plugin.cpp:198]: Cite OV if borrowed code.

File: imu_integrator/plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+				prop_data.erase(prop_data.begin()+i);
+				i--;
+			}
+		}
+
+		return prop_data;
+	}
+
+	// If an integration time ever falls inbetween two imu measurements, interpolate to it
+	static imu_type interpolate_imu(const imu_type imu_1, imu_type imu_2, double timestamp) {
```
Comment (mhuzai) [inline imu_integrator/plugin.cpp:244]: Cite OV if borrowed code.

File: install_deps.sh [hunk 4]
```diff
@@ -87,6 +87,10 @@ then
 		. ./scripts/install_openxr.sh
 	fi
 
+	if [ ! -d "${temp_dir}/gtsam" ] && y_or_n "Next: Install gtsam from souce"; then
```
Comment (mhuzai) [inline install_deps.sh:90]:
```suggestion
	if [ ! -d "${temp_dir}/gtsam" ] && y_or_n "Next: Install gtsam from source"; then
```

**Review by mhuzai — COMMENTED** [review]
1. Why are audio and hologram being reverted to older commits?
2. Merge with master and update config, includes, etc. for the new runner.

File: imu_integrator/plugin.cpp [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+		// counter++;
+#ifndef NDEBUG
+		if (input_values->last_cam_integration_time > last_cam_time) {
+			std::cout << "&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&\n";
+			last_cam_time = input_values->last_cam_integration_time;
+		}
+		pim_->resetIntegrationAndSetBias(imu_bias);
+#endif
+
+		// Predict that cam integration takes .015 seconds
```
[the remark this answers is not in the document]
Comment (JeffreyZh4ng) [reply imu_integrator/plugin.cpp:147]: Removing this is a left over comment from testing

**Review by mhuzai — COMMENTED** [review]
Finally!

**Review by apache8080 — APPROVED** [review]
LGTM

**Pull request merged**